### ISSUE TITLE
Fallback/Secondary DNS

### ISSUE THREAD

**Issue by Generator** [celzero/rethink-app#211]
## Description
RethinkDNS app only uses one DNS server, in case of service issues connections will fail.  
Some users have self-hosted DNS server (AdGuard Home, Pi-Hole ...), sometimes the server need maintenance or system fail, during maintenance users could use the main DNS.

## Solution  
Having a secondary DNS or fallback will solve the issues in case of main server failure.

**Comment by ignoramous** [comment]
Makes sense. We do plan to introduce the concept of _default server_ to which the connections would fallback upon.

That said, for DNSCrypt and DoH, servers are bootstrapped with multiple IPs where available. For example, RethinkDNS' servers are behind three Cloudflare anycast IPs and all three _can_ and _will_ be used in the current implementation.

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
> Some users have self-hosted DNS server (AdGuard Home, Pi-Hole ...), 

Would be nice to use this as default, but if can't connect for whatever reason or when you are not connected at home wifi to automatically fallback to a backup dns.

**Comment by alextenev** [comment]
Hi, Any chance for this feature to be developed?

**Comment by ignoramous** [comment]
What security reasons do you think warrant removal of a fallback? I can see the privacy angle.

Perhaps, we should add *None* as an option that disables fallback.

**Comment by ignoramous** [comment]
> security as in ensuring DNS queries ONLY go through my existing DNS firewall (NextDNS)

`None` has been there as "Fallback" for some 8 months now.

> Fallback is rarely needed so, a range of configuration options aren't really required. Besides, fallback DNS entries are hard-coded into the app and cannot be user modified.

https://github.com/golang/go/issues/8877#issuecomment-1105743745

**Comment by asl97** [comment]
https://github.com/celzero/rethink-app/commit/15ac9f9f180bab7564583100796d8c0319c56060

Why was `None` renamed to `System`?

Does this means if my chosen DNS fail, it go through my ISP's?

That's what `System` would probably mean for most users.

**Comment by ignoramous** [comment]
> Why was None renamed to System?

It has always meant *System*. Literally choosing a *None* DNS would mean loss of all connectivity.

> Does this means if my chosen DNS fail, it go through my ISP's

No.

Since `v055o`+, there's no "fallback". *Configure -> Network -> Choose fallback DNS* is only incorrectly labelled as such. It is merely a "Bootstrap DNS", that Rethink uses for its own internal use (and it *has* to be different from one user chooses in *Configure -> DNS -> Global*).

"Fallback DNS" (Bootstrap DNS, really) can be used for *trusted* apps, domains, & IPs, if user sets it so from *Configure -> DNS -> DNS for bypass*.

**Comment by asl97** [comment]
<img width="312" height="45" alt="Image" src="https://github.com/user-attachments/assets/28eed8fe-65f6-43d8-afb9-3bfd88f54c71" />

If there's no fallback anymore, you have another misleading confusing error message, it says using fallback DNS. `V0.5.5y (fdroid)`

> loss of all connectivity.

That's literally the whole original point of the option.

It's the reason why I am here, I was going to make an issue about the lack of option to disable the fallback, I search through the issue and found there was suppose to be a `None` option, which resulted in me searching for it's removal, to find the confusing name change.

But anyway, If fallback is no longer a thing and the label no longer reflect it's current function, it really should be changed and the error message amended.

TLDR: Take this as a feedback that there is a confusing chain of error and option labels

**Comment by ignoramous** [comment]
> Take this as a feedback that there is a confusing chain of error and option labels

Thanks. Please track: #2913
